timed/tracking/serializers.py

```diff
@@ -153,6 +153,41 @@ def validate(self, data):
         if not self.instance or billed is None:
             data["billed"] = task.project.billed
 
+        current_employment = Employment.objects.get_at(user=user, date=date.today())
+
+        if (
+            Task.objects.filter(
+                Q(
+                    task_assignees__user=user,
+                    task_assignees__is_reviewer=True,
+                )
+                | Q(
+                    project__project_assignees__user=user,
+                    project__project_assignees__is_reviewer=True,
+                )
+                | Q(
+                    project__customer__customer_assignees__user=user,
+                    project__customer__customer_assignees__is_reviewer=True,
+                )
+                | Q(
+                    task_assignees__user=user,
+                    task_assignees__is_manager=True,
+                )
+                | Q(
+                    project__project_assignees__user=user,
+                    project__project_assignees__is_manager=True,
+                )
+                | Q(
+                    project__customer__customer_assignees__user=user,
+                    project__customer__customer_assignees__is_manager=True,
+                )
+            ).exists()
+            and self.context["request"].method == "POST"
```

Comment:
I'm not sure the `POST` check here is required.

Also, if you reorder the conditions so that the `is_external` check comes first, Python can shortcut the expression and avoid the query if user is internal.

Reply:
`POST` is here, because only external `is_resource` should be able to create new reports, reviewers should only update them.
The error message is wrong there.